datastore-v1-proto-client/src/test/java/com/google/datastore/v1/client/it/ITDatastoreProtoClientTest.java

```diff
@@ -19,32 +19,110 @@
+  private static final String DATABASE_ID = "test-db";
 
   @Before
   public void setUp() throws GeneralSecurityException, IOException {
     DATASTORE = DatastoreHelper.getDatastoreFromEnv();
+    DATASTORE_WITH_NAMED_DB = DatastoreHelper.getDatastoreFromEnv(DATABASE_ID);
+  }
+
+  @Test
+  public void testDatastoreOptionsSimpleReadWrite() throws DatastoreException {
+    Query.Builder queryBuilder =
+        Query.newBuilder()
+            .addKind(
+                KindExpression.newBuilder()
+                    .setName("Task" + UUID.randomUUID().toString().substring(0, 10))
+                    .build());
+
+    RunQueryRequest request =
+        RunQueryRequest.newBuilder()
+            .setProjectId(PROJECT_ID)
+            .setPartitionId(PartitionId.newBuilder().build())
+            .setReadOptions(ReadOptions.newBuilder().build())
+            .setQuery(queryBuilder)
+            .build();
+
+    RunQueryResponse runQueryResponse = DATASTORE.runQuery(request);
+    List<EntityResult> entityResultsList = runQueryResponse.getBatch().getEntityResultsList();
+    Truth.assertThat(entityResultsList).isNotNull();
+  }
+
+  @Test
+  public void testDatastoreOptionsSimpleReadWriteWithNamedDb() throws DatastoreException {
+    Query.Builder queryBuilder =
+        Query.newBuilder()
+            .addKind(
+                KindExpression.newBuilder()
+                    .setName("Task" + UUID.randomUUID().toString().substring(0, 10))
+                    .build());
+
+    RunQueryRequest request =
+        RunQueryRequest.newBuilder()
+            .setProjectId(PROJECT_ID)
+            .setDatabaseId(DATABASE_ID)
+            .setPartitionId(PartitionId.newBuilder().build())
+            .setReadOptions(ReadOptions.newBuilder().build())
+            .setQuery(queryBuilder)
+            .build();
+
+    RunQueryResponse runQueryResponse = DATASTORE_WITH_NAMED_DB.runQuery(request);
+    List<EntityResult> entityResultsList = runQueryResponse.getBatch().getEntityResultsList();
+    Truth.assertThat(entityResultsList).isNotNull();
+  }
+
+  @Test
+  public void testDatastoreOptionsDbMismatch() throws DatastoreException {
+    Query.Builder queryBuilder =
+        Query.newBuilder().addKind(KindExpression.newBuilder().setName("Person").build());
+
+    RunQueryRequest request =
+        RunQueryRequest.newBuilder()
+            .setProjectId(PROJECT_ID)
+            .setDatabaseId("new-db")
+            .setPartitionId(PartitionId.newBuilder().build())
+            .setReadOptions(ReadOptions.newBuilder().build())
+            .setQuery(queryBuilder)
+            .build();
+
+    try {
+      DATASTORE.runQuery(request);
+      Assert.fail("should have thrown error");
+    } catch (IllegalArgumentException e) {
+      Truth.assertThat(e.getMessage())
+          .isEqualTo(
+              "database ids mismatched: request database id: new-db, DatastoreOptions db: null");
+    }
```

Comment:
nit: can be simplified using `org.junit.Assert#assertThrows(java.lang.Class<T>, org.junit.function.ThrowingRunnable)`.